### PR TITLE
Fix rare failure modes of monitoring test

### PR DESCRIPTION
**Pull request by tpdownes**
Address 3 intermittent issues with integration test 3:

- timeout on startup-script completion (change 5 minutes to 10 minutes)
- failure to create an Ansible tmp directory under /home when /home is mounted over NFS during startup-script
- rare occasions when ghpc binary is over-written during execution because of multiple invocations of `create_deployment.sh` in a single build

### Submission Checklist

* [X] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [X] Are all tests passing? (`make tests`)
* [X] Have you written unit tests to cover this change?
* [X] Is unit test coverage still above 80%?
* [X] Have you updated all applicable documentation?
* [X] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: tools/cloud-build/daily-tests/create_deployment.sh [hunk 1]
```diff
@@ -39,7 +41,12 @@ cd "$ROOT_DIR" ||
 		echo "*** ERROR: failed to access root directory ${ROOT_DIR} when creating blueprint"
 		exit 1
 	}
-make
+
+if [[ $ALWAYS_RECOMPILE != "no" || ! -f ghpc ]]; then
```
Comment (nick-stroud) [inline tools/cloud-build/daily-tests/create_deployment.sh:45]: nit: is it possible to pass `$ALWAYS_RECOMPILE` as a bool instead of a string?

**Review by nick-stroud — APPROVED** [review]

File: tools/cloud-build/daily-tests/create_deployment.sh (hunk 1 above)
Comment (tpdownes) [reply tools/cloud-build/daily-tests/create_deployment.sh:45]: I gave thought to using a bash boolean but didn't want to spend time handling edge cases like invocation at shell (which shell?) and conversion of YAML value in playbook to bash environment.

**Comment by tpdownes** [comment]
For posterity: the Group 1 check failed for a reason unrelated to this commit (Slurm appears not to have started up before `sinfo` command is invoked to list partitions)

**Review by nick-stroud — APPROVED** [review]
